mountpoint-s3/src/cli.rs

```diff
@@ -606,6 +606,10 @@ where
     tracing::debug!("{:?}", args);
 
     validate_mount_point(&args.mount_point)?;
+    #[cfg(feature = "sse_kms")]
+    {
+        validate_sse_args(&args.sse, &args.sse_kms_key_id)?;
```

Comment:
I was thinking of embedding this check in `CliArgs` with clap API, but seems that it is just not expressive enough for this use case.

The intention is to make `--sse_kms_key_id` to conflict with a specific value of `--sse`. The closest API clap has is [conflicts_with](https://docs.rs/clap/latest/clap/struct.Arg.html#method.conflicts_with), which makes it possible to restrict specifying `--sse_kms_key_id` when `--sse` is present. If there was an API `conflicts_with_if_eq` (similar to an existing [required_if_eq](https://docs.rs/clap/latest/clap/struct.Arg.html#method.required_if_eq)), than the check would be super neat.

But for now [custom validation](https://docs.rs/clap/latest/clap/_derive/_tutorial/chapter_3/index.html#custom-validation) as a last resort seems right.

Reply:
This is important context that should be in the code as comments.

Reply:
makes sense, added